File: cpp/1550_three_consecutive_odds/three_consecutive_odds.cpp
#include <cassert>
#include <tuple>
#include <vector>

using namespace  std;

class Solution {
public:
    bool threeConsecutiveOdds(vector<int>& arr) {
        int n = arr.size();
        for (int i = 0; i <= n - 3; ++i) {
            if ((arr[i] & 1) & (arr[i + 1] & 1) & (arr[i + 2] & 1)) {
                return true;
            }
        }
        return false;
    }
};

int main() {
  vector<tuple<vector<int>, bool>> tests {
		{{2, 6, 4, 1}, false},
		{{1, 2, 34, 3, 4, 5, 7, 23, 12}, true},  };

    for (auto &[arr, ans] : tests) {
      assert(Solution().threeConsecutiveOdds(arr) == ans);
    }
}

File: rust/1550_three_consecutive_odds/three_consecutive_odds.rs
struct Solution;

impl Solution {
    pub fn three_consecutive_odds(arr: Vec<i32>) -> bool {
        let n = arr.len();
        if n < 3 {
            return false;
        }
        for i in 0..=n - 3 {
            if (arr[i] & 1) != 0 && (arr[i + 1] & 1) != 0 && (arr[i + 2] & 1) != 0 {
                return true;
            }
        }
        return false;
    }
}

fn main() {
    let tests = vec![
        (vec![2, 6, 4, 1], false),
        (vec![1, 2, 34, 3, 4, 5, 7, 23, 12], true),
    ];

    for (arr, ans) in tests {
        assert_eq!(Solution::three_consecutive_odds(arr), ans);
    }
}
